File: sbe/binance/spot_3_1/generated-php/spot_sbe/SelfTradePreventionMode.php
<?php
/**
 * Generated SBE (Simple Binary Encoding) message codec.
 */

class SelfTradePreventionMode
{
    public const NONE = 1;
    public const EXPIRETAKER = 2;
    public const EXPIREMAKER = 3;
    public const EXPIREBOTH = 4;
    public const DECREMENT = 5;
    public const NONREPRESENTABLE = 254;
    public const NULL_VALUE = 255;

    public static function encodedLength(): int
    {
        return 1;
    }
}

File: sbe/binance/spot_3_2/generated-python/spot_sbe/SelfTradePreventionMode.py
"""Generated SBE (Simple Binary Encoding) message codec."""

import struct
from typing import List, Optional, Tuple, Set
from io import BytesIO

class SelfTradePreventionMode:
    """Enum values."""
    NONE = 1
    EXPIRE_TAKER = 2
    EXPIRE_MAKER = 3
    EXPIRE_BOTH = 4
    DECREMENT = 5
    TRANSFER = 6
    NON_REPRESENTABLE = 254
    NULL_VALUE = 255

    @staticmethod
    def encoded_length() -> int:
        return 1
